src/test/moment/duration_invalid.js

```diff
@@ -0,0 +1,79 @@
+import { module, test } from '../qunit';
+import moment from '../../moment';
+
+module('invalid');
+
+test('invalid duration', function (assert) {
+    var m = moment.duration(NaN); // should be invalid
+    assert.equal(m.isValid(), false);
+    assert.ok(isNaN(m.valueOf()));
+});
+
+test('invalid duration - only smallest unit can have decimal', function (assert) {
+    var m = moment.duration({'days': 3.5, 'hours': 1.1}); // should be invalid
+    assert.equal(m.isValid(), false);
+    assert.ok(isNaN(m.valueOf())); // .valueOf() returns NaN for invalid durations
+});
+
+test('valid duration - smallest unit can have decimal', function (assert) {
+    var m = moment.duration({'days': 3, 'hours': 1.1}); // should be valid
+    assert.equal(m.isValid(), true);
+    assert.equal(m.asHours(), 73.1);
+});
+
+test('invalid duration with two arguments', function (assert) {
+    var m = moment.duration(NaN, 'days');
+    assert.equal(m.isValid(), false);
+    assert.ok(isNaN(m.valueOf()));
+});
+
+test('invalid duration operations', function (assert) {
+    var invalids = [
+            moment.duration(NaN),
+            moment.duration(NaN, 'days')
+        ],
+        i,
+        invalid,
+        valid = moment.duration();
+
+    for (i = 0; i < invalids.length; ++i) {
+        invalid = invalids[i];
+
+        assert.ok(!invalid.add(5, 'hours').isValid(), 'invalid.add is invalid');
```

Comment:
For loops like this -- include the `i` into the string description (so when it fails you can debug it), or scrap the loop altogether -- this is unit test, you check first what is invalid, and then you check what invalid means, you don't have to test `end-to-end`.